torchaudio/csrc/ffmpeg/stream_reader/stream_processor.cpp

```diff
@@ -82,7 +84,12 @@ int StreamProcessor::process_packet(AVPacket* packet) {
       return send_frame(NULL);
     if (ret < 0)
       return ret;
-    send_frame(pFrame1);
+
+    if (pFrame1->pts >= discard_before_pts) {
+      send_frame(pFrame1);
```

Comment:
Actually, I think that the discard must happen after the frame goes through FilterGraph here https://github.com/pytorch/audio/blob/main/torchaudio/csrc/ffmpeg/stream_reader/sink.cpp#L84

StreamReader pass the AVFrame to FilterGraph before converted to Tensor, and FilterGraph can be stateful depending on the type of filter. For example, when changing FPS with [`fps` filter](https://ffmpeg.org/ffmpeg-filters.html#fps-1), a frame before the target time might be duplicated to show later which exceeds the target time.

See the following for the process flow. 
https://github.com/pytorch/audio/tree/main/torchaudio/csrc/ffmpeg#streamreader-architecture